### PR TITLE
Virualizer admin integration

### PR DESCRIPTION
**Pull request by wyf-1116**
The first step has been finished. Now the virtualizer is running with the chassis of OMAGServerApplication.

### CODE REVIEW

File: open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/AdminResource.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * VirtualizerAdminResource provides the spring annotations for the server-side
+ * implementation of the administrative interface for
+ * the Virtualizer
+ */
+
+@RestController
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/AdminResource.java]: The virtualizer should not need a RestController as it should run in the OMAG Server Chassis

File: open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * VirtualizerAdminResource provides the spring annotations for the server-side
+ * implementation of the administrative interface for
+ * the Virtualizer
+ */
+
+@RestController
+@RequestMapping("/open-metadata/admin-services/users/{userId}/servers/{serverName}")
+public class VirtualizerAdminResource {
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java]: This class should be in a new module for the virtualizer under open-metadata-implementation/governance-servers.

Comment (innerTide) [reply open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java]: Hi Mandy, the module has been added. Thanks!

**Comment by planetf1** [comment]
I took the PR to try out as I'm keen to get virtualizer running.
However when I run OMAGApplication whilst I see many services available I couldn't see any reference to the virtualizer?

**Comment by innerTide** [comment]
> I took the PR to try out as I'm keen to get virtualizer running.
> However when I run OMAGApplication whilst I see many services available I couldn't see any reference to the virtualizer?

Hi,

This PR is the first step for the integration of admin service. I created some properties for virtualiser in the corresponding directory. The full integration will be finished after the Xmas and new year holiday. Please leave it here and I will ask for reviewing and merging after the full integration.

Have a nice holiday!

Regards,

Yue Feng

File: open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/Virtualizer.java [hunk 3]
```diff
@@ -2,20 +2,54 @@
 /* Copyright Contributors to the ODPi Egeria project. */
 package org.odpi.openmetadata.virtualdataconnector.virtualiser;
 
-
-import org.springframework.beans.factory.annotation.Autowired;
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/Virtualizer.java]: This file should be removed when the virtualizer is moved into the server chassis as all of its code should be in a module under governance-servers

File: open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java]: This class should be in admin-services-spring

File: open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+     * Help the client discover the type of the server
+     */
+
+
+    /**
+     * Return the origin of this server implementation.
+     *
+     * @return Server Origin
+     */
+    @RequestMapping(method = RequestMethod.GET, path = "/server-origin")
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-demos/virtual-data-connector/virtualizer/src/main/java/org/odpi/openmetadata/virtualdataconnector/virtualiser/admin/VirtualizerAdminResource.java]: The VirtualizerResource should only be adding the extra REST calls it needs to set up the VirtualizerConfig.
The general admin services should not be repeated here.

**Pull request closed without merging**